libraries/Microsoft.Bot.Builder.Core.Extensions/ShowTypingMiddleware.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    public class ShowTypingMiddleware : IMiddleware
+    {
+        /// <summary>
+        /// (Optional) initial delay before sending first typing indicator. Defaults to 500ms.
+        /// </summary>
+        private readonly int _delay;
+
+        /// <summary>
+        /// (Optional) rate at which additional typing indicators will be sent. Defaults to every 2000ms.
+        /// </summary>
+        private readonly int _freqency;
+
+        public ShowTypingMiddleware(int delay = 500, int frequency = 2000)
+        {
```

Comment:
argument / range checking.

Reply:
@cleemullins I am not sure what any argument / range checking gets us here.  The middleware accepts int params which are passed directly to the Timer object which in turn takes int params, each with default values.  I guess we could arbitrarily check that the values are 'reasonable', e.g. not specifying a 10 minute initial delay, but all this does is present the chance of run time errors for the user, whereas left as it is I think the code will compile and run correctly even if no typing indicators are ever sent.  

What were you thinking specifically?

Reply:
@cleemullins ah. wait. non-negative check would be good :) Ill update shortly.

Reply:
@cleemullins checks added to ensure delay is >= 0 and frequency > 0.
Hopefully that should be everything.

Reply:
perfect.